### PR TITLE
Improve decimal related errors

### LINKED ISSUE #3699: Misleading error for large integers
Consider the following program:
```ballerina
import ballerina.lang.system;

function main (string[] args) {
    int a = 55555555555555555555555010100807;
    system:println(a);
}
```

Error thrown is:
```
ballerina: for input string: "55555555555555555555555010100807"
```

### PR DESCRIPTION
**Pull request by Shan1024**
## Purpose
This PR improves error messages which are shown when the decimal is too large.

Resolves #3699

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java [hunk 1]
```diff
@@ -3038,16 +3042,65 @@ private String getNodeValue(ParserRuleContext ctx, TerminalNode node) {
...
+                if (nodeValue.startsWith("-")) {
+                    dlog.error(pos, DiagnosticCode.OCTAL_TOO_SMALL, nodeValue);
+                } else {
+                    dlog.error(pos, DiagnosticCode.OCTAL_TOO_LARGE, nodeValue);
+                }
+            }
         } else if (integerLiteralContext.BinaryIntegerLiteral() != null) {
-            return Long.parseLong(getNodeValue(simpleLiteralContext, integerLiteralContext.BinaryIntegerLiteral())
-                    .toLowerCase().replace("0b", ""), 2);
+            String nodeValue = getNodeValue(simpleLiteralContext, integerLiteralContext.BinaryIntegerLiteral());
```
Comment (SupunS) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java]: This code block seems similar for all four cases. Shall we refactor this and bring to a common method, and reuse that method inside the four if-statements?

**Comment by hasithaa** [comment]
@nadeeshaan Can you please check, whether this PR has an impact on LS side. ?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java [hunk 2]
```diff
@@ -2088,10 +2095,10 @@ public void exitSimpleLiteral(BallerinaParser.SimpleLiteralContext ctx) {
...
         Set<Whitespace> ws = getWS(ctx);
-        Long longObject;
+        Object value;
         BallerinaParser.IntegerLiteralContext integerLiteralContext = ctx.integerLiteral();
-        if (integerLiteralContext != null && (longObject = getIntegerLiteral(ctx, ctx.integerLiteral())) != null) {
-            this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, longObject);
+        if (integerLiteralContext != null && (value = getIntegerLiteral(ctx, ctx.integerLiteral())) != null) {
+            this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.INT, value);
         } else if ((node = ctx.FloatingPointLiteral()) != null) {
             this.pkgBuilder.addLiteralValue(pos, ws, TypeTags.FLOAT, Double.parseDouble(getNodeValue(ctx, node)));
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java]: Shouldn't we do the same for Double as well?

Comment (Shan1024) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangParserListener.java]: This PR is intended to fix the issue in error msgs related to longs. We can fix the issue in double in another PR. WDYT?

**Pull request merged**